packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,539 @@
+   */
+  get vsize(): number {
+    return this.originalTx.virtualSize();
+  }
+
+  /**
+   * Gets the weight of the transaction in weight units.
+   * @returns {number} The weight of the transaction
+   */
+  get weight(): number {
+    return this.originalTx.weight();
+  }
+
+  /**
+   * Gets the analyzed inputs of the transaction.
+   * @returns {TransactionInput[]} An array of transaction inputs
+   */
+  get inputs(): TransactionInput[] {
+    if (!this._inputs) {
+      this._inputs = this.analyzeInputs();
+    }
+    return this._inputs;
+  }
+
+  /**
+   * Gets the analyzed outputs of the transaction.
+   * @returns {TransactionOutput[]} An array of transaction outputs
+   */
+  get outputs(): TransactionOutput[] {
+    if (!this._outputs) {
+      this._outputs = this.analyzeOutputs();
+    }
+    return this._outputs;
+  }
+
+  /**
+   * Calculates and returns the fee of the transaction in satoshis.
+   * @returns {string} The transaction fee in satoshis
+   */
+  get fee(): string {
+    if (!this._fee) {
+      this._fee = this.calculateFee();
+    }
+    return this._fee.toString();
+  }
+
+  /**
+   * Calculates and returns the fee rate of the transaction in satoshis per vbyte.
+   * @returns {string} The transaction fee rate in satoshis per vbyte
+   */
+  get feeRate(): string {
+    if (!this._feeRate) {
+      this._feeRate = new BigNumber(this.fee).dividedBy(this.vsize);
+    }
+    return this._feeRate.toString();
+  }
+
+  /**
+   * Determines if the transaction is eligible for RBF (Replace-By-Fee).
+   * @returns {boolean} True if the transaction can be replaced using RBF, false otherwise
+   */
+  get canRBF(): boolean {
+    if (this._canRBF === null) {
+      this._canRBF = this.canPerformRBF();
+    }
+    return this._canRBF;
+  }
+
+  /**
+   * Determines if the transaction is eligible for CPFP (Child-Pays-For-Parent).
+   * @returns {boolean} True if CPFP can be performed on this transaction, false otherwise
+   */
+  get canCPFP(): boolean {
+    if (this._canCPFP === null) {
+      this._canCPFP = this.canPerformCPFP();
+    }
+    return this._canCPFP;
+  }
+
+  /**
+   * Recommends the optimal fee bumping strategy based on the current transaction state.
+   * @returns {FeeBumpStrategy} The recommended fee bumping strategy
+   */
+  get recommendedStrategy(): FeeBumpStrategy {
+    if (!this._recommendedStrategy) {
+      this._recommendedStrategy = this.recommendStrategy();
+    }
+    return this._recommendedStrategy;
+  }
+
+  /**
+   * Gets the list of available UTXOs for potential use in fee bumping.
+   * @returns {UTXO[]} An array of available UTXOs
+   */
+  get getAvailableUTXOs(): UTXO[] {
+    return this.availableUTXOs;
+  }
+
+  /**
+   * Gets the current target fee rate in satoshis per vbyte.
+   * @returns {number} The target fee rate
+   */
+  get targetFeeRate(): number {
+    return this._targetFeeRate;
+  }
+
+  /**
+   * Calculates and returns the minimum fee rate required for a successful RBF.
```

Comment:
Could you provide a link for reference here. It'll be related to "incremental relay fee" I'm sure.

Reply:
Added in the new class :)